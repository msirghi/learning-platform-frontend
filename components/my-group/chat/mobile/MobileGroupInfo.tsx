import React, { useState } from 'react';
import { studentsMock } from '../../../../mocks/studentsMock';
import { StudentRow } from '../../info/StudentRow';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import { GroupInfo } from '../../info/GroupInfo';
import Fab from '@material-ui/core/Fab';
import InsertCommentIcon from '@material-ui/icons/InsertComment';
import { newsMock } from '../../../../mocks/newsMocks';
import { NewsRow } from '../../news/NewsRow';
import { Button } from '@material-ui/core';
import Fade from '@material-ui/core/Fade';
import { useTranslation } from '../../../../i18n';

type Props = {
  openChatHandler: () => void;
};

export const MobileGroupInfo: React.FC<Props> = ({ openChatHandler }) => {
  const [showMembers, setShowMembers] = useState(true);
  const { t } = useTranslation();

  const toggleShowMembers = () => setShowMembers(!showMembers);

  return (
    <div>
      <div className={styles.studentsContainer}>
        <div className={styles.chatInfoContainer}>
          <div className={styles.newsTitle}>{t('myGroup:news')}</div>
          {newsMock.map(({ id, description, date, title }, idx) => {
            return (
              <>
                <NewsRow
                  description={description}
                  key={id}
                  date={date}
                  author={'Nicolae'}
                  title={title}
                />
                {idx !== newsMock.length - 1 && <hr className={styles.separator} />}
              </>
            );
          })}
        </div>

        <Button color='primary' variant='outlined' fullWidth className={styles.seeMoreButton}>
          {t('myGroup:seeMore')}
        </Button>

        <GroupInfo
          onChevronClick={toggleShowMembers}
          chevronDown={showMembers}
          showChevron
          totalMembers={7}
          totalOnline={3}
          groupName='Name'
        />

        {showMembers && (
          <div className={styles.studentsContainer}>
            {studentsMock.map(({ firstName, lastName, id }) => {
              return <StudentRow key={id} name={`${firstName} ${lastName}`} />;
            })}
          </div>
        )}

        <Fade in>
          <div className={styles.chatButton}>
            <Fab
              color='primary'
              aria-label='add'
              className={styles.chatButton}
              onClick={() => openChatHandler()}
            >
              <InsertCommentIcon />
            </Fab>
          </div>
        </Fade>
      </div>
    </div>
  );
};
