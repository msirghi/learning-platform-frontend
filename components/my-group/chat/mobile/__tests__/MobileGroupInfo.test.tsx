import { Fab } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { MobileGroupInfo } from '../MobileGroupInfo';

describe('MobileGroupInfo component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<MobileGroupInfo openChatHandler={jest.fn()} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fire prop spy on opening the chat', () => {
    const spy = jest.fn();
    const wrapper = shallow(<MobileGroupInfo openChatHandler={spy} />);

    const fab = wrapper.find(Fab);
    expect(fab).toHaveLength(1);

    fab.simulate('click');
    wrapper.update();
    expect(spy).toBeCalled();
  });
});
