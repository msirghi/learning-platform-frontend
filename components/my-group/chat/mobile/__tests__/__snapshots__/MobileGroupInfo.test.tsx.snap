// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileGroupInfo component should match the snapshot 1`] = `
<div>
  <div
    className="studentsContainer"
  >
    <div
      className="chatInfoContainer"
    >
      <div
        className="newsTitle"
      >
        myGroup:news
      </div>
      <NewsRow
        author="Nicolae"
        date={2000-11-22T00:00:00.000Z}
        description="Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in."
        key="1"
        title="News 1"
      />
      <hr
        className="separator"
      />
      <NewsRow
        author="Nicolae"
        date={2000-11-22T00:00:00.000Z}
        description="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae."
        key="2"
        title="News 2"
      />
    </div>
    <WithStyles(ForwardRef(Button))
      className="seeMoreButton"
      color="primary"
      fullWidth={true}
      variant="outlined"
    >
      myGroup:seeMore
    </WithStyles(ForwardRef(Button))>
    <GroupInfo
      chevronDown={true}
      groupName="Name"
      onChevronClick={[Function]}
      showChevron={true}
      totalMembers={7}
      totalOnline={3}
    />
    <div
      className="studentsContainer"
    >
      <StudentRow
        key="1"
        name="First Last"
      />
      <StudentRow
        key="2"
        name="Nicolae 1"
      />
      <StudentRow
        key="3"
        name="Nicolae 2"
      />
      <StudentRow
        key="4"
        name="Aron Pumnul"
      />
      <StudentRow
        key="5"
        name="First Last"
      />
      <StudentRow
        key="6"
        name="First Last"
      />
      <StudentRow
        key="7"
        name="First Last"
      />
    </div>
    <ForwardRef(Fade)
      in={true}
    >
      <div
        className="chatButton"
      >
        <WithStyles(ForwardRef(Fab))
          aria-label="add"
          className="chatButton"
          color="primary"
          onClick={[Function]}
        >
          <Memo([object Object]) />
        </WithStyles(ForwardRef(Fab))>
      </div>
    </ForwardRef(Fade)>
  </div>
</div>
`;
